dagshub/data_engine/client/loaders/torch.py

```diff
@@ -18,13 +18,42 @@ def __init__(self, *args, **kwargs):
         self.type = "torch"
 
 
+class _BaseDataLoaderIter:
+    def __next__(self) -> Any:
+        return self.post_hook(super().__next__())
+
+
+class _SingleProcessDataLoaderIter(
+    _BaseDataLoaderIter, torch.utils.data.dataloader._SingleProcessDataLoaderIter
+):
+    def __init__(self, *args, post_hook, **kwargs):
+        self.post_hook = post_hook
+        super().__init__(*args, **kwargs)
+
+
+class _MultiProcessingDataLoaderIter(
+    _BaseDataLoaderIter, torch.utils.data.dataloader._MultiProcessingDataLoaderIter
+):
+    def __init__(self, *args, post_hook, **kwargs):
+        self.post_hook = post_hook
+        super().__init__(*args, **kwargs)
+
+
 class PyTorchDataLoader(torch.utils.data.DataLoader):
-    def __init__(self, *args, **kwargs):
+    def __init__(self, *args, post_hook=lambda x: x, **kwargs):
         super().__init__(*args, **kwargs)
+        self.post_hook = post_hook
```

Comment:
What's the purpose of the post hook?
I don't see any code in here that has custom post-hooks, so I assume this is accomodating for something you wrote yourself.
Can you add some docs?